enhancements/cloud-integration/openstack/cloud-provider-config-upgrade.md

```diff
@@ -0,0 +1,441 @@
+by the legacy cloud provider which are not supported by OpenStack CCM. Details
+of how we will handle the four removed configuration directives are provided
+below.
+
+<dl>
+<dt>secret-name</dt>
+<dt>secret-namespace</dt>
+<dt>kubeconfig-path</dt>
+<dd>
+  These are no longer relevant or supported in CCM. They define the cloud
+  credentials that are used by the legacy cloud provider. CCCMO will
+  automatically replace these with static config pointing to the cluster-managed
+  cloud credentials.
+  <br>
+  It is possible that the secret contains more information that the cloud
+  credentials populated by the installer. If so, the updated configuration will
+  no longer include these manual edits. Adding additional information to the
+  cloud credentials secret is not supported as we do not test or document it.
+  <br>
+  We estimate that the false positive risk of attempting to validate all
+  possible OpenStack authentication configurations is significantly higher than
+  the likelihood of any user actually having done this. We therefore judge that
+  it is safer not to attempt to handle this situation explicitly, and simply
+  constrain the user to modify their configuration to use the cluster cloud
+  credentials. If any user had put non-credential information in the credentials
+  secret, this would also be a prompt to them to fix that.
+</dd>
+<dt>node-volume-attach-limit</dt>
+<dd>
+  Along with all block storage options, this is now handled by Cinder CSI.
+  Unlike other block storage options, this is the only one that has been removed
+  and therefore the only one that will cause parse issues if included in
+  configuration. We will drop all block storage options when transforming the
+  user-provided configuration.
+</dd>
+</dl>
+
+We will always add `use-clouds=true` and `cloud="openstack"` in the generated
+`cloud-controller-manager-config`.  This will refer to the cluster-managed cloud
+credentials. If the user specifies these values in the user-managed configmap,
+they will be ignored and overridden.
+
+If the user specifies configuration that would not parse after transformation,
+we will mark CCCMO as `Upgradable=False`, and will not update the transformed
+configuration.
+
+### Open Questions [optional]
+
+1. After the upgrade to 4.11, we should give the user a warning if their
+   configuration is being transformed. The user should get a warning if they
+   are using static configuration such as `secret-name` and `secret-namespace`
+   and should be advised to remove this configuration. They should also see
+   warnings if they have any block-storage-related config options defined as
+   these are now handled by cinder-csi.
+
+2. Something needs to populate
+   `openshift-config-managed/cloud-controller-manager-config` before CCM can
+   run. In a running cluster CCCM0 would do this. However, CCM starts _before_
+   CCCMO which presents a bootstrapping problem on initial installation. How
```

Comment:
Wait, is that true? Looking at code it seemed to me that it's CCCMO is deploying OCCM.

Reply:
@JoelSpeed does this dependency problem go away if we're using CCCMO?

Reply:
CCCMO is provisioning CCM, so yes, this statement is nonsense.

There's not an issue here AFAICT if CCCMO is the one doing the configmap management

Reply:
Dropped this